src/components/YearbookSales/SalesBox/SalesBox.js

```diff
@@ -6,10 +6,21 @@ import PropType from "prop-types";
 
 class SalesBox extends React.Component {
 
+    constructor(props) {
+        super(props);
+        this.state = {
+            chartDataFetched: false
+        }
+    }
+
     componentDidMount() {
         this.props.actions.fetchSales();
     }
 
+    componentWillReceiveProps() {
+        this.setState({chartDataFetched: true})
```

Comment:
This will be always try call setState when component rerender. This should be control in redux app state, you can create new reducer for example loaders and dispatch action `setLoadForSales(false) ` in `fetchSales `saga.